packages/vega-scenegraph/src/util/text.js

```diff
@@ -129,10 +129,14 @@ export function offset(item) {
   // perform our own font baseline calculation
   // why? not all browsers support SVG 1.1 'alignment-baseline' :(
   var baseline = item.baseline,
-      h = fontSize(item);
+      h = fontSize(item),
+      lh = lineHeight(item);
+
   return Math.round(
+    baseline === 'line-top' ? 0.79.h + (lh-h) / 2.0 :
```

Comment:
Which would be preferred: `'line-top'` or `'top-line'`? The first reads better to me, but the second groups better with the existing options. I don't have a strong opinion, but wanted to at least consider the question.

I also have a (very slight) concern that we are adding new baseline types that are not part of existing web standards. Our current values are a subset of valid [Canvas textBaseline options](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/textBaseline). Initially, my hope was to avoid this custom offset calculation altogether, but adding them made a huge improvement in text layout consistency between SVG and Canvas (and, by extension, hopefully for any other future renderers).

Reply:
`line-top` is similar to CSS properties like `padding-top` and `line-height`. Since it also reads easier, I'd prefer `line-top` over `top-line`.